beamer/agent/chain.py

```diff
@@ -99,7 +105,15 @@ def _thread_func(self) -> None:
         self._call_on_sync_done()
         self._log.info("Sync done")
         while not self._stop:
-            events = fetcher.fetch()
+            try:
+                events = fetcher.fetch()
+                if self._latest_rpc_status is False:
+                    self._latest_rpc_status = True
+                    self._call_on_rpc_status_change(True)
+            except (ConnectionError, HTTPError):
```

Comment:
We should always try to put the minimum necessary amount of code into `try...except` blocks.
Here, the `if` statement, with it's included block, cannot throw these exceptions so we should move them out of the `try...except` block.

Reply:
I was thinking this would be a suggestion to me to do that actually to optimize otherwise i would be constantly changing the flag, the first thinking was just update all the time and changed to this one to reduce the number of calls actually

Reply:
How about this?

```python
            try:
                events = fetcher.fetch()
            except (ConnectionError, HTTPError):
                was_working = self._rpc_working:
                self._rpc_working = False
                if was_working:           
                    self._call_on_rpc_status_change(False)
            else:
                was_working = self._rpc_working:
                self._rpc_working = True
                if not was_working:           
                    self._call_on_rpc_status_change(True)
```

Reply:
well except part didnt change in some view but you moved the first part to else, i am ok with this